.ci/azure-pipelines/docs-pipeline.yaml

```diff
@@ -1,46 +1,64 @@
-trigger:
-  paths:
-    include:
-    - doc
+trigger: none
 
-pr:
-  paths:
-    include:
-    - doc
+pr: none
 
 resources:
   pipelines:
+    - pipeline: Main-CI
+      source: Main
+      trigger:
+        stages:
+        - documentation
     - pipeline: Build-CI
       source: Build
       trigger:
         stages:
         - build_gcc
   containers:
-    - container: fmt # for formatting.yaml
-      image: pointcloudlibrary/fmt
     - container: doc # for documentation.yaml
       image: pointcloudlibrary/doc
     - container: env1804 # for tutorials.yaml
       image: pointcloudlibrary/env:18.04
 
 stages:
-  - stage: formatting
-    displayName: Formatting
-    # if docs pipeline triggered by build_gcc stage,
-    # the formatting stage has already run, thus it
-    # won't run for a second time here.
-    condition: ne(variables['Build.Reason'], 'ResourceTrigger')
-    jobs:
-      - template: formatting.yaml
-
   - stage: documentation
     displayName: Documentation
-    condition: in(dependencies.formatting.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
     jobs:
       - template: documentation.yaml
 
   - stage: tutorials
     displayName: Tutorials
-    condition: in(dependencies.documentation.result, 'Succeeded', 'SucceededWithIssues')
+    dependsOn: []
     jobs:
+      - job: TutorialsDiff
+        displayName: Tutorial Source Code Diff
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - checkout: self
+          - script: | 
+              # Specify tutorial source code path
+              DIFF_SPECIFY_PATH="doc/tutorials/content/sources"
+
+              # Check the CI build reason: Pull Request or Git Push
+              BUILD_REASON=$(Build.Reason)
+              if [ "${BUILD_REASON}" -eq "PullRequest" ]; then
+                MERGE_BASE_PAIR="HEAD origin/master"
+              else
+                MERGE_BASE_PAIR="HEAD HEAD~1"
+              fi
+
+              # Initiate HasTutorialCodeChanges variable
+              echo "##vso[task.setvariable variable=HasTutorialCodeChanges;isOutput=true]false"
+
+              # Show git diff results of specified path
+              cd $(Build.SourcesDirectory)
+              git diff $(git merge-base ${MERGE_BASE_PAIR}) -- ${DIFF_SPECIFY_PATH}
+
+              # Set HasTutorialCodeChanges to true, if there are tutorial source codes changed. 
+              git diff --quiet $(git merge-base ${MERGE_BASE_PAIR}) -- ${DIFF_SPECIFY_PATH} || \
+              echo "##vso[task.setvariable variable=HasTutorialCodeChanges;isOutput=true]true"
+            name: GitResult
+            displayName: Show Git Diff Result
```

Comment:
Will this be enhanced further?

Reply:
Definitely yes! :grinning: This part is only to check if there's **some tutorial code changes** in `doc/tutorials/content/sources`. And I'll add a trigger for tutorial build, which is able to detect **PCL source code changes**.